File: app/src/main/java/com/example/mobile6/model/Medicine.java
package com.example.mobile6.model;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(tableName = "medicines")
public class Medicine {
    @NonNull
    @PrimaryKey
    private String id;
    
    private String name;
    private String ingredients;
} 

File: app/src/main/java/com/example/mobile6/domain/model/Medicine.kt
package com.example.mobile6.domain.model

import java.io.Serializable

data class Medicine(
    val id: Long,
    val name: String,
    val imgUrl: String,
    val description: String,
    val use: String,
    val usage: String,
    val precaution: String,
    val ingredients: List<Ingredient> = emptyList()
) : Serializable

data class Ingredient(
    val id: Long,
    val name: String,
    val amount: String
) : Serializable